awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsApplicationSignalsCustomizerProvider.java

```diff
@@ -61,27 +74,73 @@ public class AwsApplicationSignalsCustomizerProvider
+  private static final String DEPRECATED_APP_SIGNALS_ENABLED_CONFIG =
+      "otel.aws.app.signals.enabled";
   private static final String APPLICATION_SIGNALS_ENABLED_CONFIG =
       "otel.aws.application.signals.enabled";
-  private static final String SMP_EXPORTER_ENDPOINT_CONFIG = "otel.aws.smp.exporter.endpoint";
-  private static final String APP_SIGNALS_EXPORTER_ENDPOINT_CONFIG =
+  private static final String APPLICATION_SIGNALS_RUNTIME_ENABLED_CONFIG =
+      "otel.aws.application.signals.runtime.enabled";
+  private static final String DEPRECATED_SMP_EXPORTER_ENDPOINT_CONFIG =
+      "otel.aws.smp.exporter.endpoint";
+  private static final String DEPRECATED_APP_SIGNALS_EXPORTER_ENDPOINT_CONFIG =
       "otel.aws.app.signals.exporter.endpoint";
   private static final String APPLICATION_SIGNALS_EXPORTER_ENDPOINT_CONFIG =
       "otel.aws.application.signals.exporter.endpoint";
 
+  private static final String OTEL_JMX_TARGET_SYSTEM_CONFIG = "otel.jmx.target.system";
+
   public void customize(AutoConfigurationCustomizer autoConfiguration) {
+    autoConfiguration.addPropertiesCustomizer(this::customizeProperties);
+    autoConfiguration.addResourceCustomizer(this::customizeResource);
     autoConfiguration.addSamplerCustomizer(this::customizeSampler);
     autoConfiguration.addTracerProviderCustomizer(this::customizeTracerProviderBuilder);
+    autoConfiguration.addMeterProviderCustomizer(this::customizeMeterProvider);
     autoConfiguration.addSpanExporterCustomizer(this::customizeSpanExporter);
   }
 
   private boolean isApplicationSignalsEnabled(ConfigProperties configProps) {
     return configProps.getBoolean(
         APPLICATION_SIGNALS_ENABLED_CONFIG,
         configProps.getBoolean(
-            APP_SIGNALS_ENABLED_CONFIG, configProps.getBoolean(SMP_ENABLED_CONFIG, false)));
+            DEPRECATED_APP_SIGNALS_ENABLED_CONFIG,
+            configProps.getBoolean(DEPRECATED_SMP_ENABLED_CONFIG, false)));
+  }
+
+  private boolean isApplicationSignalsRuntimeEnabled(ConfigProperties configProps) {
+    return isApplicationSignalsEnabled(configProps)
+        && configProps.getBoolean(APPLICATION_SIGNALS_RUNTIME_ENABLED_CONFIG, true);
+  }
+
+  private Map<String, String> customizeProperties(ConfigProperties configProps) {
+    if (isApplicationSignalsRuntimeEnabled(configProps)) {
+      List<String> list = configProps.getList(OTEL_JMX_TARGET_SYSTEM_CONFIG);
+      if (list.contains("jvm")) {
+        logger.log(Level.INFO, "Found jmx in {0}", OTEL_JMX_TARGET_SYSTEM_CONFIG);
+        return Collections.emptyMap();
+      } else {
+        logger.log(Level.INFO, "Configure jmx in {0}", OTEL_JMX_TARGET_SYSTEM_CONFIG);
+        List<String> jmxTargets = new ArrayList<>(list);
+        jmxTargets.add("jvm");
+        Map<String, String> propsOverride = new HashMap<>(1);
+        propsOverride.put(OTEL_JMX_TARGET_SYSTEM_CONFIG, String.join(",", jmxTargets));
```

Comment:
q: with AppSignals enabled, if customer sets OTEL_JMX_TARGET_SYSTEM_CONFIG with value such as `kafka` , what will be AppSignals behavior?  will we OTel Gatherer also collector `kafka` metrics but dropped in CWAgent eventually?

Reply:
Yes. They will be dropped.

Metrics are not exported directly in CWA. The metrics will be further transformed in CWA, and some of them will be added a new attribute `Telemetry.Source` to be selectable by EMF exporter. Metrics not declared in the transformation rules will be dropped in the end.